pkg/registries/registries.go

```diff
@@ -24,6 +60,23 @@ type mirrorSet struct {
 	mirrorSourcePolicy config.MirrorSourcePolicy
 }
 
+// newMirror constructs an object with proper source and mirror name structures to be used in containerd registry config
+func newMirrorSet(srcImage string, mirrorLocations []config.ImageMirror, resolveTags bool,
+	mirrorSourcePolicy config.MirrorSourcePolicy) mirrorSet {
```

Comment:
consider passing down the whole entry (idms.Spec.ImageDigestMirrors) and let the function use what it needs from it, it will reduce the number of arguments and make it scale as needed.

Reply:
This wouldn't work as the function is used for ImageTagMirrors as well (and it's not worth it to introduce a generic type here)